### PR TITLE
1.2 improve playlist check file exists

### PR DESCRIPTION
**Pull request by igorshevach**

### CODE REVIEW

File: tests/unit/PlaylistGenerator-spec.js [hunk 1]
```diff
@@ -265,7 +272,7 @@ describe('Playlist Generator spec', function() {
                 return updatePlaylist(plGen,fis).then(function(playlist){
                     var playlist2 = new Playlist('test2',playlist);
                     checkKeyFrames(playlist2);
-                    done();
+                    validatePlaylistGen(plGen,done);
                 });
             })).to.eventually.be.fullfilled;
```
Comment (lilachmaliniak) [inline tests/unit/PlaylistGenerator-spec.js]: replace to.eventually.be.fullfilled with catch

File: lib/playlistGenerator/ConcatSource.js [hunk 2]
```diff
@@ -678,41 +673,48 @@ Object.defineProperty(ConcatSource.prototype , "KeyFrameDurations", {
...
                     } else {
-                        playlistUtils.dbgAssert(d === that.inner.lastKeyFrameDuration);
                         that.keyFrameDurations.last.push(that.inner.lastKeyFrameDuration);
                     }
                     delete that.inner.lastKeyFrameDuration;
                 }
                 playlistUtils.dbgAssert(that.inner.durations.length === that.keyFrameDurations.length);
             } else {
-                playlistUtils.dbgAssert(keyFrameDurations.length === 0 || idx > idxlast);
+                playlistUtils.dbgAssert(keyFrameDurations.length === 0 || idxKeyFrame > idxKeyFrameBeforeIteration);
```
Comment (lilachmaliniak) [inline lib/playlistGenerator/ConcatSource.js]: rename suggested: 
idxKeyFrameBeforeIteration ->lastKeyFrameIndex
idxKeyFram -> keyFrameIndex

File: lib/playlistGenerator/ConcatSource.js [hunk 3]
```diff
@@ -44,7 +36,11 @@ function ConcatSource(loggerInfo, playlistObj, parentObj, track, chunkPath) {
 
     if(! Object.keys(this.inner).length) {
```
Comment (igorshevach) [inline lib/playlistGenerator/ConcatSource.js]: accepted

File: lib/playlistGenerator/ConcatSource.js (hunk 2 above)
Comment (igorshevach) [inline lib/playlistGenerator/ConcatSource.js]: accepted. will refactor this code ASAP

**Pull request merged**